tidy3d/plugins/design/method.py

```diff
@@ -23,335 +20,780 @@ class Method(Tidy3dBaseModel, ABC):
     name: str = pd.Field(None, title="Name", description="Optional name for the sweep method.")
 
     @abstractmethod
-    def run(self, parameters: Tuple[ParameterType, ...], fn: Callable) -> Tuple[Any]:
-        """Defines the search algorithm (sequential)."""
+    def _run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable) -> Tuple[Any]:
+        """Defines the search algorithm."""
 
     @abstractmethod
-    def sample(self, parameters: Tuple[ParameterType, ...], **kwargs) -> Dict[str, Any]:
-        """Defines how the design parameters are sampled."""
+    def get_run_count(self, parameters: list = None) -> int:
```

Comment:
might make sense for this to be private and have some way to call it from the `DesignSpace` passing parameters, similar to `DesignSpace.run()` and `Method._run()`?

Reply:
I've made it private but I feel like including it in DesignSpace like that just short circuits `DesignSpace.summarize()`, which is calling it and printing the result.

Reply:
yea potentially. if nothing else it might just help separate some of the logic out, but I'll leave it up to you